src/datatrove/pipeline/media/media_readers/warc.py

```diff
@@ -0,0 +1,139 @@
+import heapq
+import threading
+from concurrent.futures import FIRST_COMPLETED, ThreadPoolExecutor, wait
+
+from loguru import logger
+from warcio.archiveiterator import ArchiveIterator
+
+from datatrove.data import Document, DocumentsPipeline, Media
+from datatrove.io import DataFolderLike, get_datafolder
+from datatrove.pipeline.base import PipelineStep
+
+
+class WarcReaderFast(PipelineStep):
```

Comment:
could this one not inherit BinaryReaderThreaded? (I didn't look too closely, just from a quick look)